### PR TITLE
Accept group shortnames in addition to group ids in w3c.json

### LINKED ISSUE #107: Support group shortname in w3c.json in addition to group id
The definition of w3c.json is being [amended to support either shortnames or ids](https://github.com/w3c/w3c.github.io/pull/104) for the `group` field - the repo validator should support both

### PR DESCRIPTION
**Pull request by dontcallmedom**
close #107

### CODE REVIEW

**Comment by dontcallmedom** [comment]
@deniak I don't know your level of familiarity with the validate-repo code, but would you be OK with reviewing this?

**Comment by deniak** [comment]
> @deniak I don't know your level of familiarity with the validate-repo code, but would you be OK with reviewing this?

Sure, I'll try to do it today.

File: lib/validator.js [hunk 1]
```diff
@@ -110,7 +110,21 @@ function validateRepo(r, data, licenses) {
       reportError('incompletew3cjson', {error: "group"});
     } else {
       // Note that `data.groups` is unused here.
```
Comment (deniak) [inline lib/validator.js:112]:
```suggestion
      // Note that `data.group` is unused here.
```

File: lib/validator.js [hunk 2]
```diff
@@ -110,7 +110,21 @@ function validateRepo(r, data, licenses) {
...
-      groups = arrayify(conf.group).map(id => parseInt(id, 10));
+      // conf.groups can be either a list of numeric ids
```
Comment (deniak) [inline lib/validator.js:113]:
```suggestion
      // conf.group can be either a list of numeric ids
```

File: lib/validator.js [hunk 3]
```diff
@@ -110,7 +110,21 @@ function validateRepo(r, data, licenses) {
...
+      // or a list of strings à la `${grouptype}/${groupshortname}`
+      // cf https://github.com/w3c/validate-repos/issues/107
+      groups = [];
+      for (const groupid of arrayify(conf.group)) {
+        let id = groupid;
+        if (typeof id === "string") {
+          id = parseInt(groupid, 10);
+          // this wasn't a stringified number
+          if (isNaN(id)) {
+            id = w3cgroups.find(g => g.fullshortname === groupid).id;
```
Comment (deniak) [inline lib/validator.js:123]: this should probably check that the group is found and if not, report an `inconsistentgroups` error.

**Review by deniak — CHANGES_REQUESTED** [review]

File: lib/validator.js (hunk 3 above)
Comment (dontcallmedom) [reply lib/validator.js]: 3c51e1a addresses with, but with an invalidw3cjson instead

**Review by deniak — APPROVED** [review]

**Comment by deniak** [comment]
@dontcallmedom, I see this wasn't the case before, but should validate-repo also check repository from the [functions](https://api.w3.org/functions/)?

**Comment by dontcallmedom** [comment]
re functions, let's raise a separate issue.

**Pull request merged**